### ISSUE TITLE
Some question about the fps conversion function implementation 

### ISSUE THREAD

**Issue by getupforone** [facebookresearch/SlowFast#106]
At decode() function at decoder.py, this function call   get_start_end_idx() function with some argument. 
Among the argument , second argument(clip_size) seems wrong.
 the second argument was clip_size. and this was given 'num_frames * sampling_rate * fps / target_fps'.

clip_size = num_frames * sampling_rate * fps / target_fps

in that equation 'num_frames * sampling_rate' means the number of frame before sampling. 

by divide fps with actual frame number , we can get clip_duration time(seconds).

clip_time = num_frames * sampling_rate /fps

and by multiply the clip_time with target_fps, we can get correct clip_size.

In summary,  To change the fps to target fps, we need to divide with fps and multiply target_fps.

clip_size = num_frames * sampling_rate *  target_fps / fps 

this will be correct code for this part.

    start_idx, end_idx = get_start_end_idx(
        frames.shape[0],                                    #seq_size
        num_frames * sampling_rate * target_fps / fps,      #clip_size
        clip_idx,                                           #clip_idx
        num_clips,                                          #num_clips
    )

**Comment by anuar12** [comment]
Related question: can you explain how conversion to target fps happens? I have some videos which are lower than target fps, which I would assume have to duplicated/interpolated to higher fps but I don't see this code anywhere.
Should I just convert everything to 30 fps with ffmpeg beforehand?

**Comment by takatosp1** [comment]
Hi, 
Thanks for playing with PySlowFast and dive deep into the decoding part
Basically we want to sample `num_frames * sampling_rate` frames with the target_fps. So it means we want to sample `num_frames * sampling_rate / target_fps * fps` with the fps of the video. 

One example: 
If we want a target fps of 1, and the video has a fps of 10, then we want to sample `num_frames * sampling_rate * 10` frames from the video

Looking forward to see your follow ups :)
